File: tests/benchmark/corpus/go/safe/safe_data_exfil_user_input_echo.go
// DATA_EXFIL safe: plain attacker-controlled user input forwarded to a
// fixed-destination http.Post body must not fire. Sensitivity-gate
// strips the cap because the source is Plain-tier user input.
package fixture

import (
	"net/http"
	"strings"
)

func forwardUserInput(r *http.Request) {
	msg := r.FormValue("msg")
	body := strings.NewReader(msg)
	http.Post("https://analytics.internal/track", "text/plain", body)
}

File: tests/benchmark/corpus/c/safe/safe_data_exfil_user_input_echo.c
// DATA_EXFIL safe: plain user input via fgets/stdin forwarded into the
// CURLOPT_POSTFIELDS body of a fixed-URL curl request must not fire.
// Sensitivity-gate strips the cap for Plain-tier sources.
#include <curl/curl.h>
#include <stdio.h>

void forward_stdin(void) {
    char input[256];
    if (!fgets(input, sizeof(input), stdin)) return;

    CURL *curl = curl_easy_init();
    curl_easy_setopt(curl, CURLOPT_URL, "https://telemetry.internal/forward");
    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, input);
    curl_easy_perform(curl);
    curl_easy_cleanup(curl);
}
